comp/logs/agent/config/auto_multiline_samples.go

```diff
@@ -0,0 +1,14 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package config
+
+// AutoMultilineSample defines a sample used to create auto multiline detection
+// rules
+type AutoMultilineSample struct {
```

Comment:
Are we able to use the existing [`UserSamples` struct](https://github.com/DataDog/datadog-agent/blob/main/pkg/logs/internal/decoder/auto_multiline_detection/user_samples.go#L22) instead of creating a new one? 

also - note that `matchThreshold` is not exposed in this new struct. Probably worth also exposing here.